default/python/AI/ColonisationAI.py

```diff
@@ -1053,12 +1029,75 @@ def evaluate_planet(planet_id, mission_type, spec_name, detail=None):
         if existing_presence:
             detail.append("preexisting system colony")
             retval = (retval + existing_presence * get_defense_value(spec_name)) * 2
-        if thrt_factor < 1.0:
-            retval *= thrt_factor
-            detail.append("threat reducing value by %3d %%" % (100 * (1 - thrt_factor)))
+        if threat_factor < 1.0:
+            threat_factor = revise_threat_factor(threat_factor, retval, this_sysid, MINIMUM_COLONY_SCORE)
+            retval *= threat_factor
+            detail.append("threat reducing value by %3d %%" % (100 * (1 - threat_factor)))
     return retval
 
 
+def revise_threat_factor(threat_factor, planet_value, system_id, min_planet_value=MINIMUM_COLONY_SCORE):
+    """
+    Check if the threat_factor should be made less severe.
+
+    If the AI does have enough total miltary to secure this system, and the target has more than minimal value,
+    don't let the threat_factor discount the adjusted value below min_planet_value +1, so that if there are no
+    other targets the AI could still pursue this one.  Otherwise, scoring pressure from
+    MilitaryAI.get_preferred_max_military_portion_for_single_battle might prevent the AI from pursuing a heavily
+    defended but still obtainable target even if it has no softer locations available.
+
+    :param threat_factor: the base threat factor
+    :type threat_factor: float
+    :param planet_value: the planet score
+    :type planet_value: float
+    :param system_id: the system ID of subject planet
+    :type system_id: int
+    :param min_planet_value: a floor planet value if the AI has enough military to secure the system
+    :type min_planet_value: float
+    :return: the (potentially) revised threat_factor
+    :rtype: float
+    """
+
+    # the default value below for fleetThreat shouldn't come in to play, but is just to be absolutely sure we don't
+    # send colony ships into some system for which we have not evaluated fleetThreat
+    system_status = foAI.foAIstate.systemStatus.get(system_id, {})
+    system_fleet_treat = system_status.get('fleetThreat', 1000)
+    sys_total_threat = system_fleet_treat + system_status.get('monsterThreat', 0) + system_status.get('planetThreat', 0)
+    if (MilitaryAI.get_concentrated_tot_mil_rating() > sys_total_threat) and (planet_value > 2 * min_planet_value):
+        threat_factor = max(threat_factor, (min_planet_value + 1) / planet_value)
+    return threat_factor
```

Comment:
At the very least I should have written up my thinking there.   So let's talk through my recollection of my thinking, see if you agree about it and then we either document that or adjust the handling.

Keep in mind that when this code kicks in, it simply acts to preserve a minimally-valued target that will only be pursued if there are essentially no other targets of this type; the alternative is that the target is just absolutely ignored even if there are no other targets.  If we totally ignore this target, then depending on its location relative to empire holdings then the presence of the surrounding area threat might not even be in range to triggered a heightened Military demand (and associated priority for building additional military ships).  So my thinking was that leaving it as a minimally-valued target could help insure that the associated threat leads to increased military construction.

Of course, if there are not any truly good colony opportunities (and therefore low colony priority) then most excess production capacity is likely going into military anyways, but there would still be some competition between warship construction and troopship construction, and this would give a bit more focus on the warship production.   

And also, that same argument could even be extended to say not even worry here about whether we currently have enough military to take on the direct system, but that seemed like a bit too much of a tenuous opportunity then.

But perhaps we shouldn't really put our thumb on the scale in favor of warships then, and simply let the construction proceed based on whatever priorities are justified elsewhere.  It seems like a bit of a tossup to me, I'd be fine with it either way.